eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Mgmt/src/Providers/ResourceClientProvider.cs

```diff
@@ -239,58 +242,91 @@ private MethodProvider BuildOperationMethod(InputServiceMethod method, MethodPro
                 {
                     result.Add(parameter);
                 }
             }
+
             return result;
         }
 
-        protected CSharpType GetOperationMethodReturnType(bool isAsync, bool isLongRunningOperation, IReadOnlyList<InputOperationResponse> operationResponses, out bool isGeneric)
+        protected bool IsLongRunningOperation(InputServiceMethod method)
+        {
+            return method is InputLongRunningServiceMethod || method is InputLongRunningPagingServiceMethod;
+        }
+
+        protected bool IsReturnTypeGeneric(InputServiceMethod method)
+        {
+            var operationResponses = method.Operation.Responses;
+            var response = operationResponses.FirstOrDefault(r => !r.IsErrorResponse);
+            var responseBodyType = response?.BodyType is null ? null : ManagementClientGenerator.Instance.TypeFactory.CreateCSharpType(response.BodyType);
+            return IsLongRunningOperation(method) && responseBodyType is not null;
+        }
+
+        protected CSharpType GetOperationMethodReturnType(InputServiceMethod method, bool isAsync, out bool isGeneric)
         {
-            isGeneric = false;
+            bool isLongRunningOperation = IsLongRunningOperation(method);
+            isGeneric = IsReturnTypeGeneric(method);
+
             if (isLongRunningOperation)
             {
-                var response = operationResponses.FirstOrDefault(r => !r.IsErrorResponse);
-                var responseBodyType = response?.BodyType is null ? null : ManagementClientGenerator.Instance.TypeFactory.CreateCSharpType(response.BodyType);
-                if (responseBodyType is null)
+                if (!isGeneric)
                 {
                     return isAsync ? new CSharpType(typeof(Task<>), typeof(ArmOperation)) : typeof(ArmOperation);
                 }
                 else
                 {
-                    isGeneric = true;
-                    return isAsync ? new CSharpType(typeof(Task<>), new CSharpType(typeof(ArmOperation<>), Type)) : new CSharpType(typeof(ArmOperation<>), Type);
+                    return isAsync ? new CSharpType(typeof(Task<>), new CSharpType(typeof(ArmOperation<>), ResourceClientCharpType)) : new CSharpType(typeof(ArmOperation<>), ResourceClientCharpType);
                 }
             }
-            return isAsync ? new CSharpType(typeof(Task<>), new CSharpType(typeof(Response<>), Type)) : new CSharpType(typeof(Response<>), Type);
+            return isAsync ? new CSharpType(typeof(Task<>), new CSharpType(typeof(Response<>), ResourceClientCharpType)) : new CSharpType(typeof(Response<>), ResourceClientCharpType);
         }
 
-        private TryStatement BuildOperationMethodTryStatement(MethodProvider convenienceMethod, bool isAsync, InputServiceMethod method, bool isGeneric)
+        // TODO: should we fix the generated context declaration statement format? there semicolon after a new line is weird
```

Comment:
remove this TODO, there is not any extra work in MPG plugin after MTG fixes the writer behavior